addons/mail/models/mail_activity.py

```diff
@@ -699,15 +703,18 @@ def activity_schedule(self, act_type_xmlid='', date_deadline=None, summary='', n
         if isinstance(date_deadline, datetime):
             _logger.warning("Scheduled deadline should be a date (got %s)", date_deadline)
         if act_type_xmlid:
-            activity_type = self.sudo().env.ref(act_type_xmlid)
+            try:
+                activity_type = self.env.ref(act_type_xmlid)
```

Comment:
Didn't we talk about an heuristics like
 * asked xml id
 * if not: todo
 * if not: first activity type linked to model or generic
 * if not: go to hell, keep False

Reply:
asked xml id
if not: todo
if not: first generic (model = False)